sdk/core/ts-http-runtime/review/ts-http-runtime.api.md

```diff
@@ -46,6 +32,63 @@ export interface Agent {
     sockets: unknown;
 }
 
+// @public
+export interface ApiKeyAuthScheme {
+    apiKeyLocation: ApiKeyLocation;
+    name: string;
+    type: "apiKey";
```

Comment:
Nit: do we want to use kind as the discriminatory property?

Reply:
Good q, TypeSpec uses `type` here but I know we use `kind` in other places. I don't have a strong opinion on this

Reply:
I don't know if it can be used as the discriminatory property since `BasicAuth` and `BearerAuth` share the same `type` in the definition [here](https://github.com/microsoft/typespec/blob/main/packages/http/lib/auth.tsp#L21-L53)

Reply:
I am leaning towards `kind` slightly since `type` can also be used as a keyword. But we already use both `type` and `kind` in our existing SDK API surface so no strong feeling.

Reply:
I also slightly favor `kind`, so I'll make updates accordingly unless someone has strong arguments for `type`.